src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java

```diff
@@ -133,5 +135,41 @@ public static String getMD5(String plainText) {
     public static String getWorkDirectory() {
         return Jenkins.getInstance().root.getAbsolutePath();
     }
+    
+    public static String getUserAgent() {
+        String version = null;
+        String instanceId = null;
+
+        try {
+            version = Utils.class.getPackage().getImplementationVersion();
+            instanceId = Jenkins.getInstance().getLegacyInstanceId();
+        } catch (Exception e) {
+        }
+
+        if (version == null) {
+            version = "local";
+        }
+        if (instanceId == null) {
+            instanceId = "local";
+        }
+
+        return Constants.PLUGIN_NAME + "/" + version + "/" + instanceId;
+    }
+    
+    public static OperationContext injectUserAgent(String userInfo) {
```

Comment:
I would hide the required ugliness for getting the userInfo inside the injectUserAgent.
Overload the method with different argument types for all the SDK objects like CloudBlob, CloudBlobContainer and in each of these fetch the userInfo and call the method with the string argument.
In this way you won't have to have that extra line in each call